### PR TITLE
refactor: config group to pure function and move away from group design

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

refactor 

**Did you add tests for your changes?**
Already present

**If relevant, did you update the documentation?**
NA

**Summary**
- removed redundant entries from cli flag file
- Refactored config group to pure function

**Does this PR introduce a breaking change?**
No

**Other information**

https://github.com/webpack/webpack-cli/pull/1765#discussion_r475077508

### CODE REVIEW

**Review by evenstensberg — COMMENTED** [review]
Let's keep entry default option

**Comment by anshumanv** [comment]
> Let's keep entry default option

It's unused

**Review by evilebottnawi — COMMENTED** [review]
Also default value for `entry` is invalid, so :+1:

**Comment by anshumanv** [comment]
We need to run ConfigGroup needless whether config args are present or nay, I think we should make it a pure function, this should be a good start to move away from group design.

Thoughts? @evilebottnawi @evenstensberg

**Comment by evilebottnawi** [comment]
> We need to run ConfigGroup needless whether config args are present or nay, I think we should make it a pure function, this should be a good start to move away from group design.

Agree, we don't need group design and run only necessary code

**Review by rishabh3112 — DISMISSED** [review]

**Review by evenstensberg — APPROVED** [review]

**Review by snitin315 — COMMENTED** [review]
Good job 🔥

**Review by snitin315 — APPROVED** [review]

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ const modeAlias = {
     development: 'dev',
 };
 
+let opts = {
+    outputOptions: {},
+    options: {},
+};
+
+// Return a list of default configs in various forrmats
```
Comment (jamesgeorge007) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:37]: Typo

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 2]
```diff
@@ -58,162 +64,163 @@ const getConfigInfoFromFileName = (filename) => {
...
     }
+    return newOptionsObject;
+};
 
-    async run() {
-        await this.resolveConfigFiles();
-        await this.resolveConfigMerging();
-        return this.opts;
+const resolveConfigMerging = async (args) => {
+    const { merge } = args;
```
Comment (jamesgeorge007) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:201]: Isn't it better to use destructuring if the function takes in a single argument?

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:201]: neh I'll prefer this, looks cleaner

**Review by snitin315 — APPROVED** [review]

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 3]
```diff
@@ -58,162 +64,163 @@ const getConfigInfoFromFileName = (filename) => {
...
+
+const handleConfigResolution = async (args) => {
+    const parsedArgs = arrayToObject(args);
+    await resolveConfigFiles(parsedArgs);
+    await resolveConfigMerging(parsedArgs);
+    return opts;
+};
 
-module.exports = ConfigGroup;
+module.exports = handleConfigResolution;
```
Comment (evilebottnawi) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:226]: I think we need move this from `ConfigGroup.js` file, to be honestly we don't need `groups` files :smile: Do you want to do it in this PR?

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:226]: Yep lets do it here

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:226]: Or maybe let's migrate all groups first then we can move them all from `groups` to `argResolvers`?

Comment (evilebottnawi) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:226]: Maybe we put each function in own file, anyway let's refactor firstly and when we can think about places

Comment (anshumanv) [reply packages/webpack-cli/lib/groups/ConfigGroup.js:226]: yep once all groups are migrated, we can organize it neatly

**Review by evilebottnawi — APPROVED** [review]
Good start

**Pull request merged**